lib/fs.js

```diff
@@ -1463,7 +1463,9 @@ function StatWatcher() {
   };
 
   this._handle.onstop = function() {
-    self.emit('stop');
+    process.nextTick(function() {
+      self.emit('stop');
+    });
```

Comment:
Can we avoid allocating a new closure here? When calling `stop()` on a massive tree of objects, we will be allocating a lot of closures for nothing.

``` js
this._handle.onstop = function() {
  process.nextTick(emitStop, self)
}

// to be put a root level
function emitStop(self) {
  self.emit('stop');
}
```